### PR TITLE
registry with badges, no tests added yet

### PR DESCRIPTION
**Pull request by zmitton**
the research that got us there: https://consensys.quip.com/80HRAP7Ozbux

### CODE REVIEW

**Comment by oed** [comment]
Since you choose to use byte32 instead of bytes because of gas, are you going to assume IPFS hashes starts with Qm?
Would maybe be nice with a helperfunction:
```
getSelfIssued(bytes32 registrationIdentifier, address attestee) returns(bytes32){
    return get(registrationIdentifier, attesteee, attestee);
}
```

**Comment by zmitton** [comment]
ropsten: 0x94bdf8c549986bd84b2a4abb10b23c3f59884cdf
ethereum: 0xcc39e8ab206229c95b9649ead3560e1712a72eed

**Comment by zmitton** [comment]
@oed yeah that's the assumption 'Qm'. the helper functions I decided to leave out becuause they cost more, and I figure we should be making the helpers instead at the javascript level

File: build/contracts/UportRegistry.sol.js [hunk 1]
```diff
@@ -350,18 +350,26 @@ var SolidityEvent = require("web3/lib/web3/event.js");
   "default": {
     "abi": [
       {
-        "constant": true,
+        "constant": false,
```
Comment (coder5876) [inline build/contracts/UportRegistry.sol.js]: This is the function `get`, right? Why was it changed to `"constant": false`?

File: contracts/UportRegistry.sol [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
+
+  //create or update
+  function set(bytes32 registrationIdentifier, address attestee, bytes32 value){
+      Set(registrationIdentifier, msg.sender, attestee);
+      registry[registrationIdentifier][msg.sender][attestee] = value;
   }
 
-  function getAttributes(address personaAddress) constant returns(bytes) {
-    return ipfsAttributeLookup[personaAddress];
+  function get(bytes32 registrationIdentifier, address attestor, address attestee) returns(bytes32){
```
Comment (coder5876) [inline contracts/UportRegistry.sol]: You can specify here that this function is `constant` (as we do with the `getAttributes` function).

File: build/contracts/UportRegistry.sol.js (hunk 1 above)
Comment (coder5876) [reply build/contracts/UportRegistry.sol.js]: I saw it below...

**Comment by coder5876** [comment]
Ok, this looks good @zmitton. Next thing that needs to happen is that the javascript functions `getAttributes` and `setAttributes` need to be updated to use the new contract interface. I can do this and push to this branch.

**Comment by coder5876** [comment]
Looks great! Update the version please in `package.json` to `3.0.0`, and add an entry to 
`RELEASE-NOTES.md`.

**Pull request merged**